detector/other/TrackerEndcapSupport_o1_v01.cpp

```diff
@@ -70,6 +70,13 @@ static Ref_t create_detector(Detector& theDetector, xml_h e, SensitiveDetector)
     Tube    l_tub(rmin, rmax, layerWidth, 2 * M_PI);
     Volume  l_vol(l_nam, l_tub, air);
     l_vol.setVisAttributes(theDetector, x_layer.visStr());
+
+    DetElement layer_pos(sdet, l_nam + "_pos", l_num);
+    pv = envelope.placeVolume(l_vol, Position(0, 0, zmin + layerWidth / 2.));
+    pv.addPhysVolID("layer", l_num);
+    pv.addPhysVolID("barrel", 1);
+    layer_pos.setPlacement(pv);
+
```

Comment:
what changed here? did you just move this block before the 'for'?

Reply:
Yes, so that I could assign the surface to the `layer_pos` DetElement